src/Tester/StreamingTests/EHSubscriptionMultiplicityTests.cs

```diff
@@ -0,0 +1,147 @@
+
+namespace UnitTests.StreamingTests
+{
+    [DeploymentItem("OrleansConfigurationForTesting.xml")]
+    [DeploymentItem("OrleansProviders.dll")]
+    [TestClass]
+    public class EHSubscriptionMultiplicityTests : UnitTestSiloHost
+    {
+        private const string StreamProviderName = "EventHubStreamProvider";
+        private const string StreamNamespace = "EHSubscriptionMultiplicityTestsNamespace";
+        private const string EHPath = "orleans_test";
+        private const string EHConsumerGroup = "orleanstestconsumer";
+
+        private readonly SubscriptionMultiplicityTestRunner runner;
+
+        // TODO: Get ServiceBus connection string from OrleansTestStorageKey.txt - jbragg
+        // WARNING: ServiceBus stream providers are still under developement and have not been fully wired up to testing environments, hence this hack.
+        // If you add your connection string here for testing !!1!REMOVE IT BEFORE YOU COMMIT!!!1
+        private static readonly EventHubSettings EventHubConfig = new EventHubSettings("Endpoint=Add_Your_service_servicebus_connection_string_here_BUT_DONT_LEAK_IT_IN_A_PR!!!",
+            EHConsumerGroup, EHPath);
+
+        private static readonly EventHubStreamProviderConfig ProviderConfig =
+            new EventHubStreamProviderConfig(StreamProviderName);
+
+        public EHSubscriptionMultiplicityTests()
+            : base(new TestingSiloOptions
+            {
+                StartFreshOrleans = true,
+                SiloConfigFile = new FileInfo("OrleansConfigurationForTesting.xml"),
+                AdjustConfig = AdjustClusterConfiguration
+            })
+        {
+            runner = new SubscriptionMultiplicityTestRunner(StreamProviderName, GrainClient.Logger);
+        }
+
+        public static void AdjustClusterConfiguration(ClusterConfiguration config)
+        {
+            var settings = new Dictionary<string, string>();
+            // get initial settings from configs
+            ProviderConfig.WriteProperties(settings);
+            EventHubConfig.WriteProperties(settings);
+
+            // add queue balancer setting
+            settings.Add(PersistentStreamProviderConfig.QUEUE_BALANCER_TYPE, StreamQueueBalancerType.DynamicClusterConfigDeploymentBalancer.ToString());
+
+            // register stream provider
+            config.Globals.RegisterStreamProvider<EventHubStreamProvider>(StreamProviderName, settings);
+            config.Globals.RegisterStorageProvider<MemoryStorage>("PubSubStore");
+
+            // make sure all node configs exist, for dynamic cluster queue balancer
+            config.GetConfigurationForNode("Primary");
+            config.GetConfigurationForNode("Secondary_1");
+        }
+
+        // Use ClassCleanup to run code after all tests in a class have run
+        [ClassCleanup]
+        public static void MyClassCleanup()
+        {
+            StopAllSilos();
+        }
+
+        [TestMethod, TestCategory("EventHub"), TestCategory("Streaming")]
```

Comment:
Absolutely!  We just don't have an environment for automated testing yet, so they are not ready for nightly testing.  Now that we have something we can test, we can update the test environments, then add these to functional tests.